File: Array/11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        
        int l = 0;
        int r = height.length-1;
        int maxWater = 0;

        while(l<r){
            maxWater = Math.max(Math.min(height[l], height[r]) * (r-l), maxWater);


            if(height[l] < height[r]) {
                l++;
            } else {
                r--;
            }
        }
        return maxWater;
    }
}

File: JavaScript/11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)
        l = 0
        r = n-1

        water = 0

        while l<r:
            if height[l] < height[r]:
                water = max(water,(r-l)* height[l])
                l+=1
            else:
                water = max(water,(r-l)* height[r])
                r-=1
        return water
